src/ess/polarization/base.py

```diff
@@ -308,18 +312,30 @@ def __call__(self, wavelength: sc.Variable) -> sc.Variable:
 
 def he3_opacity_from_cell_params(
     pressure: He3CellPressure[Cell],
-    cell_length: He3CellLength[Cell],
+    length: He3CellLength[Cell],
+    temperature: He3CellTemperature[Cell],
 ) -> He3OpacityFunction[Cell]:
     """
     Opacity function for a given cell, based on pressure and cell length.
 
     Note that this can alternatively be defined via neutron beam data, see
     :py:func:`he3_opacity_from_beam_data`.
     """
-    # TODO Is this the correct unit?
-    factor_from_he3_number_density = 0.07733 * sc.Unit('1/N')
-    # TODO What unit scale should we use?
-    opacity0 = factor_from_he3_number_density * pressure * cell_length
+    from scipp.constants import Boltzmann as k_B
+
+    he3_neutron_absorption_cross_section_at_1_angstrom = 2966.0e-24 * sc.Unit(
+        'cm^2/Angstrom'
+    )
+    # Try to convert to Kelvin, since we get a more cryptic error message later if we
+    # do not, in case the user passes in a temperature in degree Celsius.
+    temperature = temperature.to(unit='K')
+
+    opacity0 = (
+        he3_neutron_absorption_cross_section_at_1_angstrom
+        / (k_B * temperature)
+        * pressure
+        * length
+    )
     return He3OpacityFunction[Cell](OpacityFunction(opacity0))
 
 
```

Comment:
Have to be careful with units that will be inserted here, most commonly [bar] for pressure, and [cm] for cell length. I see you define T in [K], sigma in [cm2/Angstrom]. k_B has [m2*kg/(s2*K)]. It fits if k_B is converted to have cm instead of m. For the pressure we can assume they insert in [bar] = 10^5[Pa] = 10^5 kg/(m*s2), which then also needs to be changed to cm.

Reply:
Scipp will take care of conversions, nothing to worry about (most likely).

Reply:
It does not convert automatically in, e.g., addition, but here we multiply and divide, so all the conversion factors are just combined and taken care off.

Reply:
perfect!